go/libraries/doltcore/schema/schema.go

```diff
@@ -45,6 +46,10 @@ type Schema interface {
 
 	// SetPkOrdinals specifies a primary key column ordering
 	SetPkOrdinals([]int) error
+
+	// AddColumn adds a column to this schema in the order given and returns the resulting Schema.
+	// The new column cannot be a primary key. To alter primary keys, create a new schema with those keys.
+	AddColumn(column Column, order *sql.ColumnOrder) (Schema, error)
```

Comment:
I don't think we want `sql` abstractions at this layer of the codebase

Reply:
fixed